src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/SegmentAllocator.java

```diff
@@ -333,24 +333,62 @@ default MemorySegment allocate(long bytesSize) {
      * the allocator capacity.
      *
      * @param size the size (in bytes) of the allocation arena.
-     * @param scope the scope associated with the segments returned by this allocator.
+     * @param scope the scope associated with the segments returned by the arena-based allocator.
      * @return a new bounded arena-based allocator
      * @throws IllegalArgumentException if {@code size <= 0}.
      * @throws IllegalStateException if {@code scope} has been already closed, or if access occurs from a thread other
      * than the thread owning {@code scope}.
      */
-    static SegmentAllocator arenaAllocator(long size, ResourceScope scope) {
+    static SegmentAllocator arenaBounded(long size, ResourceScope scope) {
         Objects.requireNonNull(scope);
         return scope.ownerThread() == null ?
                 new ArenaAllocator.BoundedSharedArenaAllocator(scope, size) :
                 new ArenaAllocator.BoundedArenaAllocator(scope, size);
     }
 
     /**
-     * Returns a native unbounded arena-based allocator.
+     * Returns a native unbounded arena-based allocator, with predefined block size.
      * <p>
-     * The returned allocator allocates a memory segment {@code S} of a certain fixed size (using malloc) and then
-     * responds to allocation requests in one of the following ways:
+     * The returned allocator {@linkplain MemorySegment#allocateNative(long, ResourceScope) allocates} a memory segment
+     * {@code S} of a certain (fixed) block size and then responds to allocation requests in one of the following ways:
+     * <ul>
+     *     <li>if the size of the allocation requests is smaller than the size of {@code S}, and {@code S} has a <em>free</em>
+     *     slice {@code S'} which fits that allocation request, return that {@code S'}.
+     *     <li>if the size of the allocation requests is smaller than the size of {@code S}, and {@code S} has no <em>free</em>
+     *     slices which fits that allocation request, allocate a new segment {@code S'} (using malloc), which has same size as {@code S}
+     *     and set {@code S = S'}; the allocator then tries to respond to the same allocation request again.
+     *     <li>if the size of the allocation requests is bigger than the size of {@code S}, allocate a new segment {@code S'}
+     *     (using malloc), which has a sufficient size to satisfy the allocation request, and return {@code S'}.
```

Comment:
Still a mention of `malloc` here, as well as in the same text for the other `arenaUnbounded` overload.

Reply:
Sorry - missed those. Hopefully fixed now.